root/etc/e-smith/templates/etc/ssh/sshd_config/70Restricted2Sftp

```diff
@@ -1,23 +1,57 @@
 {
     # restrict groups to sftp, ssh access is refused
+    return "" if ($sssd{'ShellOverrideStatus'} ne 'enabled');
+
     my $domain = $DomainName;
     my @SftpRestricted = ();
+    my @SftpUnRestricted = ();
+
+    my $admins_group = $admins{'group'} || 'domain admins';
 
-    foreach ( split(',',$sshd{'AllowGroups'} || '')) {
+    foreach ( $admins_group, split(',',$sshd{'AllowGroups'} || '')) {
         my ($group, $sftp) = split(':', $_);
-        if(!$group || !$sftp) {
+        if(!$group) {
             next;
         }
+        if (!$sftp) {
+            $sftp = '';
+        }
+
         $group .= "\@$domain" if ($group !~ '@');
-        push @SftpRestricted, $group if ($sftp eq 'sftp');
+        if ($sftp eq 'sftp') {
+            push @SftpRestricted, $group;
+        } else {
+            # do not force SFTP to groups with SSH permission
+            push @SftpUnRestricted, $group;
+       }
     }
 
-    if (@SftpRestricted) {
-        my $matchGroup = join (",",@SftpRestricted);
-        $OUT .= "Match Group \"$matchGroup\"\n";
+   if ($sshd{'AllowEveryone'} eq 'sftp+ssh') {
+        # Everyone is allowed to SSH & SFTP
+        return  "";
+    } 
+    elsif (@SftpRestricted && ($sshd{'AllowEveryone'} eq 'none')) {
+        # We list non restricted group to SFTP
+        @SftpUnRestricted = join (",",@SftpUnRestricted);
+        # We list restricted group to SFTP
+        @SftpRestricted = join (",",@SftpRestricted);
+
+        $OUT .= "Match Group \"root,@SftpUnRestricted\"\n";
+        $OUT .= "    ForceCommand none\n";
+        $OUT .= "Match Group \"@SftpRestricted\"\n";
         $OUT .= "    ForceCommand internal-sftp\n";
+        $OUT .= "    AllowTCPForwarding no\n";
+        $OUT .= "    X11Forwarding no\n";
+
+    } elsif ($sshd{'AllowEveryone'} eq 'sftp') {
+        # We list non restricted group to SFTP
+        @SftpUnRestricted = join (",",@SftpUnRestricted);
+
+        $OUT .= "Match Group \"root,@SftpUnRestricted\"\n";
+        $OUT .= "    ForceCommand none\n";
```

Comment:
We must state port forwarding is enabled, otherwise `Match All` wins and disables it.

```suggestion
        $OUT .= "    ForceCommand none\n";
        $OUT .= "    AllowTCPForwarding yes\n";
        $OUT .= "    X11Forwarding yes\n";
```

Reply:
done